src/components/filter/SchulenFilter.vue

```diff
@@ -0,0 +1,226 @@
+    } else {
+      newSelection = findSchulenInResultByIds(ids);
+    }
+    organisationStore.schulenFilter.selectedItems = newSelection;
+  };
+
+  const handleSelectionUpdate = (selection: SelectedSchulenIds): void => {
+    if (isEmptySelection(selection)) {
+      updateOrganisationenIds([]);
+      updateSearchString(undefined);
+      clearInput();
+    } else {
+      updateOrganisationenIds(selection);
+    }
+    mirrorSelectionToStore(selection);
+  };
+
+  const isSameSelection = (a: SelectedSchulenIds, b: SelectedSchulenIds): boolean => {
+    return sameContent(wrapSelectedSchulenIds(a), wrapSelectedSchulenIds(b));
+  };
+
+  type SelectionChange = [SelectedSchulenIds, SelectedSchulenIds, boolean];
+  type PreviousSelectionChange = [SelectedSchulenIds, Array<string> | undefined, boolean | undefined];
+
+  watch(
+    [selectedSchulenIds, (): Array<string> => wrapSelectedSchulenIds(props.initialIds), hasAutoselectedSchule],
+    (
+      [currentSelection, currentInitialIds, currentlyHasAutoselectedSchule]: SelectionChange,
+      [oldSelection, oldInitialIds]: PreviousSelectionChange,
+    ) => {
+      const hasSelectionChanged: boolean = !isSameSelection(currentSelection, oldSelection);
+      const havePropsChanged: boolean = !isSameSelection(currentInitialIds, oldInitialIds);
+      if (hasSelectionChanged) {
+        handleSelectionUpdate(currentSelection);
+      } else if (havePropsChanged) {
+        selectedSchulenIds.value = currentInitialIds;
+        handleSelectionUpdate(currentInitialIds);
+      } else if (isEmptySelection(currentSelection)) {
+        let tempIds: SelectedSchulenIds;
+        if (currentlyHasAutoselectedSchule && autoselectedSchule.value) {
+          tempIds = unwrapSelectedSchulenIds([autoselectedSchule.value.id]);
+        } else {
+          tempIds = currentInitialIds;
+        }
+        if (!isSameSelection(selectedSchulenIds.value, tempIds)) {
+          selectedSchulenIds.value = tempIds;
+          handleSelectionUpdate(tempIds);
+        }
+      }
+    },
+    { immediate: true },
+  );
+
+  watch(
+    () => organisationStore.schulenFilter.filterResult,
+    () => {
+      const ids: SelectedSchulenIds = selectedSchulenIds.value;
+      // this is a special case, because we can't assume that the appropriate schule is in the store already
+      // this is redundant in all other cases
+      if (props.initialIds) mirrorSelectionToStore(ids);
+    },
+  );
+
+  watch(
+    schulenFilter,
+    async (newFilter: OrganisationenFilter, oldFilter: OrganisationenFilter | undefined) => {
+      if (oldFilter) {
+        clearTimeout(timerId.value);
+        if (hasAutoselectedSchule.value) return;
+        timerId.value = setTimeout(async () => {
+          await organisationStore.loadSchulenForFilter(newFilter);
+        }, 500);
+      } else {
+        // initial load
+        await organisationStore.loadSchulenForFilter(newFilter);
+      }
+    },
+    { immediate: true },
+  );
```

Comment:
```suggestion
watch(
  schulenFilter,
  async (newFilter, oldFilter) => {
    // The timer can be cleared in all cases??
    if (timerId.value) clearTimeout(timerId.value);
    
    // We skiip if we have an autoselectedSchule
    if (oldFilter && hasAutoselectedSchule.value) return;
    
    // We apply the debounce of 500 only when there is an oldFilter (a change has been made)
    const delay = oldFilter ? 500 : 0;
    
    timerId.value = setTimeout(async () => {
      await organisationStore.loadSchulenForFilter(newFilter);
    }, delay);
  },
  { immediate: true }
);
```
Hier auch leserlicher mit Kommentaren und weniger nested ifs